server/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -399,18 +399,21 @@ public SimpleQueryStringBuilder fuzzyTranspositions(boolean fuzzyTranspositions)
     protected Query doToQuery(QueryShardContext context) throws IOException {
         Settings newSettings = new Settings(settings);
         final Map<String, Float> resolvedFieldsAndWeights;
+        boolean isAllField;
         if (fieldsAndWeights.isEmpty() == false) {
             resolvedFieldsAndWeights = QueryParserHelper.resolveMappingFields(context, fieldsAndWeights);
+            isAllField = QueryParserHelper.hasAllFieldsWildcard(fieldsAndWeights.keySet());
         } else {
             List<String> defaultFields = context.defaultFields();
-            boolean isAllField = defaultFields.size() == 1 && Regex.isMatchAllPattern(defaultFields.get(0));
-            if (isAllField) {
-                newSettings.lenient(lenientSet ? settings.lenient() : true);
-            }
+            isAllField = defaultFields.size() == 1 && Regex.isMatchAllPattern(defaultFields.get(0));
```

Comment:
@jimczi Why do we check only the first element here, shouldn't we check all the `defaultFields`?
(same applies to all 3 types of queries)

Reply:
I agree we should check all elements. However I don't think it's useful to set `fields` or the default fields to something like `["*", "field1", "field2"]` but for consistency we should always set the leniency to true if `*` is part of the list of fields to query.